apps/src/lib/kits/maker/portScanning.js

```diff
@@ -30,9 +31,10 @@ export function findPortWithViableDevice() {
       .then(list => {
         const bestOption = getPreferredPort(list);
         if (bestOption) {
-          return bestOption.comName;
+          return Promise.resolve(bestOption.comName);
         } else {
-          throw new Error('Did not find a usable device on a serial port.');
+          return Promise.reject(new ConnectionFailedError(
+              'Did not find a usable device on a serial port.'));
```

Comment:
Nice! Throwing a typed exception is much better than the error message string matching I'm doing elsewhere in applab.